Bump yaml versions

elezar opened the pull request:
This change updates the sigs.k8s.io/yaml version from v1.3.0 to v1.4.0 and switches to using gopkg.in/yaml.v3 for ordered YAML output. This means that consumers no longer need to pull in two YAML dependencies (v2 and v3).

See https://github.com/containerd/containerd/pull/11522

klihub reviewed (approved):
Shall we then tag a v1.0.1 ?

elezar commented:
> Shall we then tag a v1.0.1 ?

Given that this is causing issues in containerd and buildkit (https://github.com/moby/buildkit/pull/5856) that may be useful.

elezar commented:
For `containerd` we have:
```
diff --git a/go.mod b/go.mod
index baba349dc..253387a8e 100644
--- a/go.mod
+++ b/go.mod
@@ -144,7 +144,6 @@ require (
 	golang.org/x/time v0.7.0 // indirect
 	google.golang.org/genproto/googleapis/api v0.0.0-20250115164207-1a7da9e5054f // indirect
 	gopkg.in/inf.v0 v0.9.1 // indirect
-	gopkg.in/yaml.v2 v2.4.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 	k8s.io/api v0.32.2 // indirect
 	k8s.io/apiserver v0.32.2 // indirect
@@ -168,3 +167,5 @@ exclude (
 	github.com/davecgh/go-spew v1.1.2-0.20180830191138-d8f796af33cc
 	github.com/pmezard/go-difflib v1.0.1-0.20181226105442-5d4384ee4fb2
 )
+
+replace tags.cncf.io/container-device-interface v1.0.0 => ../container-device-interface
diff --git a/go.sum b/go.sum
index e906275c3..8171b33a2 100644
--- a/go.sum
+++ b/go.sum
@@ -545,7 +545,6 @@ gopkg.in/evanphx/json-patch.v4 v4.12.0/go.mod h1:p8EYWUEYMpynmqDbY58zCKCFZw8pRWM
 gopkg.in/inf.v0 v0.9.1 h1:73M5CoZyi3ZLMOyDlQh031Cx6N9NDJ2Vvfl76EDAgDc=
 gopkg.in/inf.v0 v0.9.1/go.mod h1:cWUDdTG/fYaXco+Dcufb5Vnc6Gp2YChqWtbxRZE0mXw=
 gopkg.in/yaml.v2 v2.2.1/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
-gopkg.in/yaml.v2 v2.4.0 h1:D8xgwECY7CYvx+Y2n4sBz93Jn9JRvxdiyyo8CTfuKaY=
 gopkg.in/yaml.v2 v2.4.0/go.mod h1:RDklbk79AGWmwhnvt/jBztapEOGDOx6ZbXqjP6csGnQ=
 gopkg.in/yaml.v3 v3.0.0-20200313102051-9f266ea9e77c/go.mod h1:K4uyk7z7BCEPqu6E+C64Yfv1cQ7kz7rIZviUmN+EgEM=
 gopkg.in/yaml.v3 v3.0.1 h1:fxVm/GzAzEWqLHuvctI91KS9hhNmmWOoWu0XTYJS7CA=
@@ -578,7 +577,5 @@ sigs.k8s.io/structured-merge-diff/v4 v4.4.2 h1:MdmvkGuXi/8io6ixD5wud3vOLwc1rj0aN
 sigs.k8s.io/structured-merge-diff/v4 v4.4.2/go.mod h1:N8f93tFZh9U6vpxwRArLiikrE5/2tiu1w1AGfACIGE4=
 sigs.k8s.io/yaml v1.4.0 h1:Mk1wCc2gy/F0THH0TAp1QYyJNzRm2KCLy3o5ASXVI5E=
 sigs.k8s.io/yaml v1.4.0/go.mod h1:Ejl7/uTz7PSA4eKMyQCUTnhZYNmLIl+5c2lQPGR2BPY=
-tags.cncf.io/container-device-interface v1.0.0 h1:fbwPQiWZNpXUb9Os6t6JW52rsOppTFUbeJOpNtN1TmI=
-tags.cncf.io/container-device-interface v1.0.0/go.mod h1:mmi2aRGmOjK/6NR3TXjLpEIarOJ9qwgZjQ3nTIRwAaA=
 tags.cncf.io/container-device-interface/specs-go v1.0.0 h1:8gLw29hH1ZQP9K1YtAzpvkHCjjyIxHZYzBAvlQ+0vD8=
 tags.cncf.io/container-device-interface/specs-go v1.0.0/go.mod h1:u86hoFWqnh3hWz3esofRFKbI261bUlvUfLKGrDhJkgQ=

```
And after vendoring:
```
On branch dependabot/go_modules/tags.cncf.io/container-device-interface-1.0.0
Your branch is up to date with 'upstream/dependabot/go_modules/tags.cncf.io/container-device-interface-1.0.0'.

Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   go.mod
	modified:   go.sum
	deleted:    vendor/gopkg.in/yaml.v2/.travis.yml
	deleted:    vendor/gopkg.in/yaml.v2/LICENSE
	deleted:    vendor/gopkg.in/yaml.v2/LICENSE.libyaml
	deleted:    vendor/gopkg.in/yaml.v2/NOTICE
	deleted:    vendor/gopkg.in/yaml.v2/README.md
	deleted:    vendor/gopkg.in/yaml.v2/apic.go
	deleted:    vendor/gopkg.in/yaml.v2/decode.go
	deleted:    vendor/gopkg.in/yaml.v2/emitterc.go
	deleted:    vendor/gopkg.in/yaml.v2/encode.go
	deleted:    vendor/gopkg.in/yaml.v2/parserc.go
	deleted:    vendor/gopkg.in/yaml.v2/readerc.go
	deleted:    vendor/gopkg.in/yaml.v2/resolve.go
	deleted:    vendor/gopkg.in/yaml.v2/scannerc.go
	deleted:    vendor/gopkg.in/yaml.v2/sorter.go
	deleted:    vendor/gopkg.in/yaml.v2/writerc.go
	deleted:    vendor/gopkg.in/yaml.v2/yaml.go
	deleted:    vendor/gopkg.in/yaml.v2/yamlh.go
	deleted:    vendor/gopkg.in/yaml.v2/yamlprivateh.go
	modified:   vendor/modules.txt
	modified:   vendor/tags.cncf.io/container-device-interface/pkg/cdi/spec.go

no changes added to commit (use "git add" and/or "git commit -a")
```
Showing the removal of the `vendor/gopkg.in/yaml.v2` dependency.

elezar commented:
> Shall we then tag a v1.0.1 ?

I have created #263 to track the release.
